### PR TITLE
Default activation response to empty JsObject.

### PR DESCRIPTION
**Pull request by markusthoemmes**
When using a blocking call, reduced to return only the response the unsafe conversion creates an uncaught exception.

### CODE REVIEW

**Comment by rabbah** [comment]
This is OK - where are we using the response result though incorrectly; isn't there another bug?

**Comment by markusthoemmes** [comment]
I discovered it in a loopback invoker test where I stubbed the invoker out. Was a mere accident. I was thinking though if we should change the type of `reponse` to not be `Option`. Didn't try what that'd mean though.

**Comment by rabbah** [comment]
could be... or remove the option and use an empty object. None is used on the non-blocking path so the result doesn’t matter.

**Comment by markusthoemmes** [comment]
No PG needed here I'd say?

File: common/scala/src/main/scala/whisk/core/entity/WhiskActivation.scala [hunk 1]
```diff
@@ -97,13 +97,13 @@ case class WhiskActivation(namespace: EntityPath,
...
-        cause.map(("cause" -> _.toJson)) ++
+        cause.map("cause" -> _.toJson) ++
         actionOrNot())
   }
 
-  def resultAsJson = response.result.toJson.asJsObject
+  def resultAsJson: JsObject = response.result.map(_.toJson.asJsObject).getOrElse(JsObject.empty)
 
-  def toExtendedJson = {
+  def toExtendedJson: JsObject = {
```
Comment (rabbah) [inline common/scala/src/main/scala/whisk/core/entity/WhiskActivation.scala]: can you check this one as well

**Comment by rabbah** [comment]
Travis is enough - The result = None is not observable unless there's an invariant violation; returning result = None and a full activation instead of an id (I suppose the loopback return a WhiskActivation() here for example). 

It would be nice to tighten but I think it's equally OK to treat a missing result as `{}`... until we want to support `JsNull` as a possible response as well (related to sequence semantics).

File: common/scala/src/main/scala/whisk/core/entity/WhiskActivation.scala (hunk 1 above)
Comment (rabbah) [reply common/scala/src/main/scala/whisk/core/entity/WhiskActivation.scala]: i checked it - it's fine.

**Comment by markusthoemmes** [comment]
Closing this as abandoned.

**Pull request closed without merging**